checker/src/call_visitor.rs

```diff
@@ -559,16 +559,23 @@ impl<'call, 'block, 'analysis, 'compilation, 'tcx, E>
                     return;
                 }
 
-                // If the condition is always true, this assumption is redundant
-                if cond_as_bool.unwrap_or(false) {
-                    let span = self.block_visitor.bv.current_span;
-                    let warning =
-                        self.block_visitor.bv.cv.session.struct_span_warn(
-                            span,
-                            "assumption is provably true and can be deleted",
-                        );
-                    self.block_visitor.bv.emit_diagnostic(warning);
-                }
+                // If the condition is always true, this assumption is redundant. If false, the
+                // assumption is ignored. Otherwise, no diagnostics are emitted.
+                let message = if cond_as_bool == Some(true) {
+                    "assumption is provably true and can be deleted"
+                } else if cond_as_bool == Some(false) {
+                    "assumption is provably false and it will be ignored in the assertion"
```

Comment:
Your test case never gets here. That should be very interesting. I suspect that check_condition_value_and_reachability does the wrong thing because of the false assumption. It should be interesting to look into that.